pkg/action/executor/k8s_controller_updater.go

```diff
@@ -74,11 +74,13 @@ func newK8sControllerUpdater(clusterScraper *cluster.ClusterScraper, ormClient *
 				dynClient:           clusterScraper.DynamicClient,
 				dynNamespacedClient: clusterScraper.DynamicClient.Resource(res).Namespace(namespace),
 			},
-			name:         kind,
-			ormClient:    ormClient,
-			managerApp:   managerApp,
-			gitConfig:    gitConfig,
-			k8sClusterId: clusterId,
+			name:                  kind,
+			ormClient:             ormClient,
+			managerApp:            managerApp,
+			gitConfig:             gitConfig,
+			k8sClusterId:          clusterId,
+			gitOpsConfigCache:     *clusterScraper.GitOpsConfigCache,
+			gitOpsConfigCacheLock: &clusterScraper.GitOpsConfigCacheLock,
```

Comment:
I dont generally define a pointer variable until there is a need for one. I don't understand the need here.

Reply:
Are you referring to the lock here?

Reply:
Ignore this plz.. :-)